### ISSUE TITLE
  Failure to create a new hive source. What can be done about it?

### ISSUE THREAD

**Issue by shiliquan** [apache/incubator-dolphinscheduler#3791]
创建hive源，开启了kdc认证，本地可以hive直接登录，但是作为hive源连接的话就报如下错误，引用了hive的jiar包后，服务则无法启动，想知道报这种错误是哪里设置不对还是什么问题？要怎么解决？谢谢！
![Catch(09-23-01-07-41)](https://user-images.githubusercontent.com/42087586/93915107-19bd7a80-fd3a-11ea-8167-64631be139f3.jpg)

**Comment by zhuangchong** [comment]
Please check whether the hive jar version referenced by ds is consistent with the hive server version, and whether there are fewer jars. If you are not sure, you can copy all the hive server hive-*.jar to the ds lib directory.

---
请查看ds 引用的hive jar 版本和 hive 服务端版本是否一致，是否少jar，如果不确定可以把hive 服务端 hive-*.jar 全部拷贝到ds lib目录下

**Comment by xingchun-chen** [comment]
please update the title in English

**Comment by shiliquan** [comment]
https://github.com/apache/incubator-dolphinscheduler/issues/3791#issuecomment-697073154
我把hive-*.jar包全部拷贝到lib目录下后，重新安装后，api-server服务无法启动，页面打不开了，帮忙看看

[INFO] 2020-09-23 10:53:11.837 org.eclipse.jetty.util.log:[193] - Logging initialized @3214ms to org.eclipse.jetty.util.log.Slf4jLog
[INFO] 2020-09-23 10:53:11.927 org.eclipse.jetty.server.Server:[370] - jetty-9.4.14.v20181114; built: 2018-11-14T21:20:31.478Z; git: c4550056e785fb5665914545889f21dc136ad9e6; jvm 1.8.0_202-b08
[INFO] 2020-09-23 10:53:11.958 org.eclipse.jetty.server.handler.ContextHandler.application:[2345] - Initializing Spring embedded WebApplicationContext
[INFO] 2020-09-23 10:53:12.001 org.eclipse.jetty.server.session:[365] - DefaultSessionIdManager workerName=node0
[INFO] 2020-09-23 10:53:12.001 org.eclipse.jetty.server.session:[370] - No SessionScavenger set, using defaults
[INFO] 2020-09-23 10:53:12.003 org.eclipse.jetty.server.session:[149] - node0 Scavenging every 660000ms
[INFO] 2020-09-23 10:53:12.007 org.eclipse.jetty.server.handler.ContextHandler:[855] - Started o.s.b.w.e.j.JettyEmbeddedWebAppContext@4b87074a{application,/dolphinscheduler,[file:///tmp/jetty-docbase.7667266662126961927.9200/, jar:file:/opt/soft/dolphinscheduler/lib/swagger-bootstrap-ui-1.9.3.jar!/META-INF/resources, jar:file:/opt/soft/dolphinscheduler/lib/springfox-swagger-ui-2.9.2.jar!/META-INF/resources],AVAILABLE}
[INFO] 2020-09-23 10:53:12.008 org.eclipse.jetty.server.Server:[407] - Started @3387ms
[INFO] 2020-09-23 10:53:12.023 org.apache.dolphinscheduler.service.zk.ZookeeperOperator:[82] - zookeeper registry center init, server lists is: 172.17.9.39:2181,172.17.9.40:2181,172.17.9.41:2181.
[INFO] 2020-09-23 10:53:12.079 org.apache.curator.utils.Compatibility:[48] - Running in ZooKeeper 3.4.x compatibility mode
[INFO] 2020-09-23 10:53:12.080 org.apache.curator.utils.Compatibility:[61] - Using emulated InjectSessionExpiration
[INFO] 2020-09-23 10:53:12.156 org.apache.curator.framework.imps.CuratorFrameworkImpl:[308] - Starting
[INFO] 2020-09-23 10:53:12.181 org.apache.curator.framework.imps.CuratorFrameworkImpl:[356] - Default schema
[INFO] 2020-09-23 10:53:12.193 org.apache.curator.framework.state.ConnectionStateManager:[251] - State change: CONNECTED
[INFO] 2020-09-23 10:53:12.197 org.apache.dolphinscheduler.service.zk.ZookeeperOperator:[82] - zookeeper registry center init, server lists is: 172.17.9.39:2181,172.17.9.40:2181,172.17.9.41:2181.
[INFO] 2020-09-23 10:53:12.197 org.apache.curator.framework.imps.CuratorFrameworkImpl:[308] - Starting
[INFO] 2020-09-23 10:53:12.198 org.apache.curator.framework.imps.CuratorFrameworkImpl:[356] - Default schema
[INFO] 2020-09-23 10:53:12.199 org.apache.curator.framework.state.ConnectionStateManager:[251] - State change: CONNECTED
[INFO] 2020-09-23 10:53:12.207 org.apache.dolphinscheduler.service.zk.ZookeeperCachedOperator:[43] - add listener to zk path: /dolphinscheduler
[INFO] 2020-09-23 10:53:13.360 org.apache.dolphinscheduler.service.zk.ZookeeperOperator:[82] - zookeeper registry center init, server lists is: 172.17.9.39:2181,172.17.9.40:2181,172.17.9.41:2181.
[INFO] 2020-09-23 10:53:13.361 org.apache.curator.framework.imps.CuratorFrameworkImpl:[308] - Starting
[INFO] 2020-09-23 10:53:13.362 org.apache.curator.framework.imps.CuratorFrameworkImpl:[356] - Default schema
[INFO] 2020-09-23 10:53:13.363 org.apache.curator.framework.state.ConnectionStateManager:[251] - State change: CONNECTED
[INFO] 2020-09-23 10:53:13.364 org.apache.dolphinscheduler.service.zk.ZookeeperCachedOperator:[43] - add listener to zk path: /dolphinscheduler
[INFO] 2020-09-23 10:53:14.384 springfox.documentation.spring.web.plugins.DocumentationPluginsBootstrapper:[160] - Context refreshed
[INFO] 2020-09-23 10:53:14.399 springfox.documentation.spring.web.plugins.DocumentationPluginsBootstrapper:[163] - Found 1 custom documentation plugin(s)
[INFO] 2020-09-23 10:53:14.478 springfox.documentation.spring.web.scanners.ApiListingReferenceScanner:[41] - Scanning for api listing references
[INFO] 2020-09-23 10:53:14.524 org.apache.dolphinscheduler.remote.NettyRemotingClient:[376] - netty client closed
[INFO] 2020-09-23 10:53:14.525 org.apache.dolphinscheduler.service.log.LogClientService:[59] - logger client closed
[INFO] 2020-09-23 10:53:14.531 org.eclipse.jetty.server.session:[167] - node0 Stopped scavenging
[INFO] 2020-09-23 10:53:14.533 org.eclipse.jetty.server.handler.ContextHandler:[1045] - Stopped o.s.b.w.e.j.JettyEmbeddedWebAppContext@4b87074a{application,/dolphinscheduler,[file:///tmp/jetty-docbase.7667266662126961927.9200/, jar:file:/opt/soft/dolphinscheduler/lib/swagger-bootstrap-ui-1.9.3.jar!/META-INF/resources, jar:file:/opt/soft/dolphinscheduler/lib/springfox-swagger-ui-2.9.2.jar!/META-INF/resources],UNAVAILABLE}

**Comment by xingchun-chen** [comment]
Has the issue been resolved？

**Comment by shiliquan** [comment]
Not Resolved, is the current version does not support the connection of HIV 3?

**Comment by xingchun-chen** [comment]
Is the environment variable HIVE_HOME configured in  conf/env/dolphinscheduler_env.sh of DS
![image](https://user-images.githubusercontent.com/55787491/97851937-0508d500-1d31-11eb-991e-01512df48ebe.png)

**Comment by shiliquan** [comment]
Configured as shown in figure：
![image](https://user-images.githubusercontent.com/42087586/97852443-b60f6f80-1d31-11eb-8816-c699cbd8f348.png)

**Comment by xingchun-chen** [comment]
Need to configure the environment variables of the hive-client

**Comment by shiliquan** [comment]
What do you mean? You mean the environment variable for the current user of the system: . Yeah, this?

**Comment by xingchun-chen** [comment]
< What do you mean? You mean the environment variable for the current user of the system: . Yeah, this?
yes

**Comment by shiliquan** [comment]
修改过后还是不行，报错依旧，先确定下当前1.3.2的jar包版本是否支持hive3的数据源连接？
![image](https://user-images.githubusercontent.com/42087586/97855597-ece78480-1d35-11eb-98f8-71bff098a6ae.png)
![image](https://user-images.githubusercontent.com/42087586/97855659-0983bc80-1d36-11eb-8a61-9b59ade8f65b.png)

**Comment by xingchun-chen** [comment]
Using hive3, you can modify the source pom file, and recompile and package, please refer to this article https://my.oschina.net/dailidong/blog/4483145

**Comment by shiliquan** [comment]
The current environment is not CDH environment, and Hadoop, hive and other versions are already 3. The X version, is there any other way to solve it? No source compiled, there is no compiled good tar package can be shared under, thank you!

**Comment by xingchun-chen** [comment]
Which version of DS?

**Comment by xingchun-chen** [comment]
which version of hive3 ?
![image](https://user-images.githubusercontent.com/55787491/98098205-fd7c3400-1ec8-11eb-8c63-b300766b02c4.png)

**Comment by shiliquan** [comment]
1.3.2的版本

**Comment by xingchun-chen** [comment]
hive3需要哪个版本？

**Comment by shiliquan** [comment]
version 3.1.0.3.1.4.0-315

**Comment by xingchun-chen** [comment]
The installation package is extracted from the link below
链接: https://pan.baidu.com/s/1GQOOAALZVkbrGkuDeJnD8g 提取码: b279 
You can replace hive 2.1.0 jar in the lib directory with 3.1.0, then restart the service
![image](https://user-images.githubusercontent.com/55787491/98336884-ae5a0e80-2042-11eb-9999-adfda7697a29.png)

![image](https://user-images.githubusercontent.com/55787491/98336860-a601d380-2042-11eb-8e71-246fb64a9f3c.png)

**Comment by shiliquan** [comment]
Thank you very much. Would you like to simply replace the jar package? No need to reinstall it, right?

**Comment by xingchun-chen** [comment]
> Thank you very much. Would you like to simply replace the jar package? No need to reinstall it, right?

yes, you need restart the api/worker server

**Comment by xingchun-chen** [comment]
Wether the problem  has solved?

**Comment by shiliquan** [comment]
还没来得急测试，这两天试一下再回复你！

**Comment by shiliquan** [comment]
![image](https://user-images.githubusercontent.com/42087586/98500152-a5f21580-2286-11eb-91fc-7c409084cb98.png)

**Comment by shiliquan** [comment]
当前报错：
![image](https://user-images.githubusercontent.com/42087586/98500858-8b20a080-2288-11eb-846a-f767befa803e.png)

**Comment by shiliquan** [comment]
当前报错：
![image](https://user-images.githubusercontent.com/42087586/98500953-cd49e200-2288-11eb-8c35-837c1618b27b.png)

**Comment by shiliquan** [comment]
Thank you very much. That problem has been solved. The reason for the problem is that hive doesn't create the source link using the ticket in the DS configuration file. I set the ticket in the DS to:
![image](https://user-images.githubusercontent.com/42087586/98513809-6f76c380-22a3-11eb-80fd-36e1a84b0b4d.png)
 
However, the correct setting of the Principal value at the time of creation of the HIVE source should be consistent with the Principal value of the source host, otherwise it will fail:
![image](https://user-images.githubusercontent.com/42087586/98514006-c2e91180-22a3-11eb-892e-1ca5e1504af4.png)
